nav2_dynamic_motion_model/include/nav2_dynamic_motion_model/motion_model_interface.hpp

```diff
@@ -0,0 +1,34 @@
+#ifndef NAV2_DYNAMIC_MOTION_MODEL__MOTION_MODEL_INTERFACE_HPP_
+#define NAV2_DYNAMIC_MOTION_MODEL__MOTION_MODEL_INTERFACE_HPP_
+
+#include <memory>
+#include <geometry_msgs/msg/pose.hpp>
+#include <geometry_msgs/msg/pose_array.hpp>
+#include <geometry_msgs/msg/twist.hpp>
+#include <nav2_dynamic_msgs/msg/obstacle.hpp>
+#include <nav2_dynamic_msgs/msg/obstacle_array.hpp>
+
+namespace nav2_dynamic_motion_model
+{
+
+class MotionModelInterface
```

Comment:
Like we talked about, I think we should make sure to include `initialize()` methods so that there's an obvious place to do initialization activities like creating subscriptions, getting parameters, etc. 

Think a bit about what we might want the arguments for it to be. It might be good to look at other Nav2 plugin interfaces for example. (rclcpp::node, std::string name, etc)